nsis/pcsx2_shared_init.nsi

```diff
@@ -0,0 +1,224 @@
+!include WinVer.nsh
+  ${IfNot} ${AtLeastWinVista}
+    MessageBox MB_OK "Your operating system is unsupported by PCSX2. Please upgrade your operating system or install PCSX2 1.4.0."
+    Quit
+  ${EndIf}
+
+ClearErrors
+UserInfo::GetName
+  Pop $R8
+
+UserInfo::GetOriginalAccountType
+Pop $UserPrivileges
+
+  # GetOriginalAccountType will check the tokens of the original user of the
+  # current thread/process. If the user tokens were elevated or limited for
+  # this process, GetOriginalAccountType will return the non-restricted
+  # account type.
+  # On Vista with UAC, for example, this is not the same value when running
+  # with `RequestExecutionLevel user`. GetOriginalAccountType will return
+  # "admin" while GetAccountType will return "user".
+  ;UserInfo::GetOriginalAccountType
+  ;Pop $R2
+
+${If} $UserPrivileges == "Admin"
+    StrCpy $IsAdmin 1
+    ${ElseIf} $UserPrivileges == "User"
+    StrCpy $IsAdmin 0
+${EndIf}
+FunctionEnd
+
+Function PreInstallDialog
+
+nsDialogs::Create /NOUNLOAD 1018
+Pop $PreInstall_Dialog
+
+    GetDlgItem $PreInstall_DlgBack $HWNDPARENT 3
+    EnableWindow $PreInstall_DlgBack ${SW_HIDE}
+
+    GetDlgItem $PreInstall_DlgNext $HWNDPARENT 1
+    EnableWindow $PreInstall_DlgNext 0
+
+${NSD_CreateProgressBar} 0 75 100% 10% "Test"
+    Pop $hwnd
+
+  ${NSD_CreateTimer} NSD_Timer.Callback 1
+
+nsDialogs::Show
+FunctionEnd
+
+Function NSD_Timer.Callback
+${NSD_KillTimer} NSD_Timer.Callback
+    SendMessage $hwnd ${PBM_SETRANGE32} 0 100
+
+!include WinVer.nsh
+!include "X64.nsh" 
+${If} ${AtLeastWin8.1}
+${OrIf} $IsAdmin == 0
+Call PreInstall_UsrWait
+SendMessage $HWNDPARENT ${WM_COMMAND} 1 0
+
+${EndIf}
+
+${If} ${RunningX64}
+ReadRegDword $R0 HKLM "SOFTWARE\Wow6432Node\Microsoft\VisualStudio\14.0\VC\Runtimes\x86" "Installed"
+${Else}
+   ReadRegDword $R0 HKLM "SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86" "Installed"
+${EndIf}
+    Pop $R0
+
+${If} $R0 == "1"
+Goto ExecDxSetup
+${EndIf}
+
+${NSD_CreateLabel} 0 45 100% 10u "Downloading Visual C++ 2015 package"
+inetc::get "https://download.microsoft.com/download/9/3/F/93FCF1E7-E6A4-478B-96E7-D4B285925B00/vc_redist.x86.exe" "$TEMP\vcredist_2015_Update_1_x86.exe" /SILENT /CONNECTTIMEOUT 30 /RECEIVETIMEOUT 30 /END
+    ${NSD_CreateLabel} 0 45 100% 10u "Installing Visual C++ 2015 package"
+    SendMessage $hwnd ${PBM_SETPOS} 20 0
+    ExecWait '"$TEMP\vcredist_2015_Update_1_x86.exe /S"'
+    SendMessage $hwnd ${PBM_SETPOS} 40 0
+    Delete "$TEMP\vcredist_2015_Update_1_x86.exe"
+
+ExecDxSetup:
+
+SetOutPath "$TEMP"
+SendMessage $hwnd ${PBM_SETPOS} 60 0
+${NSD_CreateLabel} 0 45 100% 10u "Installing DXWebSetup package"
+SendMessage $hwnd ${PBM_SETPOS} 80 0
+
+File "dxwebsetup.exe"
+ExecWait '"$TEMP\dxwebsetup.exe" /Q' $DirectXSetupError
+
+${NSD_CreateLabel} 0 45 100% 20u "Finished installing Visual C++ Redistributables. Click Next to continue."
+${NSD_CreateLabel} 0 45 100% 20u "Finished installing Visual C++ Redistributables. Click Next to continue."
```

Comment:
It'd be better to figure out what the bug is.

BTW, a brief look at the documentation suggests that you need to `Pop` the window handles (HWND) off the stack whenever you use `NSD_CreateLabel` (or any `NSD_Create*` function).